File: Baekjoon/Baekjoon01946.py
# Baekjoon01946.py
import sys
input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n = int(input())

    scores = []
    for _ in range(n):
        scores.append(tuple(map(int, input().split())))
    scores.sort()

    ans = 1
    minInterviewScore = scores[0][1]
    for i in range(1, n):
        if (minInterviewScore > scores[i][1]):
            ans += 1
            minInterviewScore = scores[i][1]

    print(ans)

File: Baekjoon/Baekjoon01946.cpp
// Baekjoon01946.cpp
#include <iostream>

#include <algorithm>

using namespace std;

using pii = pair<int, int>;

void solve() {
    int n;
    cin >> n;

    pii scores[100'000];
    for (int i = 0; i < n; i++)
        cin >> scores[i].first >> scores[i].second;
    sort(scores, scores + n);

    int ans = 1, minSecondScore = scores[0].second;
    for (int i = 1; i < n; i++) {
        if (minSecondScore > scores[i].second) {
            ans++;
            minSecondScore = scores[i].second;
        }
    }

    cout << ans << '\n';    
}

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int t;
    cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}